include/pulsar/Client.h

```diff
@@ -404,6 +404,19 @@ class PULSAR_PUBLIC Client {
      */
     uint64_t getNumberOfConsumers();
 
+    /**
+     * Asynchronously get the SchemaInfo of a topic and a specific version.
+     *
+     * @topic the topic name
+     * @version the schema version byte array, see Message::getLongSchemaVersion.
+     * @callback the callback that is triggered when the SchemaInfo is retrieved successfully or not.
+     *
+     * NOTE: If there is no schema registered or the given version does not exist, a SchemaInfo whose type is
```

Comment:
I think we can add a new `Result` : `TopicSchemaNotFound`

When schema is not found, the Result on the callback is `TopicSchemaNotFound`.

What do you think?

Reply:
In Java client, the `TopicNotFound` error is ignored and an empty optional of `SchemaInfo` is returned. So I thought it should not be treated as an exceptional case and users should process this error. I just use a `NONE` schema to replace an empty optional.

But I just thought again and IMO an error should be returned if we cannot get the schema for the given topic and version. So I will adopt the suggestion you made so that users should not handle the schema type.

Reply:
I won't add a new `Result`. The server side returns a `TopicNotFound` error. https://github.com/apache/pulsar/blob/f7c0b3c49c9ad8c28d0b00aa30d727850eb8bc04/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java#L999

So I use `ResultTopicNotFound` here.